src/app/clusters/level-control/level-control.cpp

```diff
@@ -139,6 +141,27 @@ static void timerCallback(System::Layer *, void * callbackContext)
     emberAfLevelControlClusterServerTickCallback(static_cast<EndpointId>(reinterpret_cast<uintptr_t>(callbackContext)));
 }
 
+static uint32_t calculateNextTimestampMs(uint32_t delayMs)
+{
+    int32_t waitTime, latency, delay = (int32_t) delayMs;
+    const chip::System::Clock::Timestamp currentTime = chip::System::SystemClock().GetMonotonicTimestamp();
+
+    latency = (int32_t) currentTime.count() - (int32_t) nextTimestamp.count();
+
+    if (latency > delay)
+    {
+        waitTime = 0;
+    }
+    else
+    {
+        waitTime = delay - latency;
+    }
+
+    nextTimestamp += chip::System::Clock::Milliseconds32(delayMs);
```

Comment:
This seems like it might be clearer if we did the arithmetic with durations and timestamps instead of converting from timestamps to integers and then back to timestamps....

Reply:
I wanted to take into account the case when the timer is faster than expected but I guess this is not really needed.